flake.nix

```diff
@@ -59,34 +59,52 @@
             activate.custom;
 
           activate = rec {
-            custom = base: activate: pkgs.buildEnv {
-              name = ("activatable-" + base.name);
-              paths = [
-                base
-                (pkgs.writeTextFile {
-                  name = base.name + "-activate-path";
-                  text = ''
-                    #!${pkgs.runtimeShell}
-                    set -euo pipefail
-
-                    ${activate}
-                  '';
-                  executable = true;
-                  destination = "/deploy-rs-activate";
-                })
-                (pkgs.writeTextFile {
-                  name = base.name + "-activate-rs";
-                  text = ''
-                    #!${pkgs.runtimeShell}
-                    exec ${self.defaultPackage."${system}"}/bin/activate "$@"
-                  '';
-                  executable = true;
-                  destination = "/activate-rs";
-                })
-              ];
-            };
-
-            nixos = base: custom base.config.system.build.toplevel ''
+            custom =
+              {
+                __toString = customSelf: "TODO: dryActivate";
+                __functor = customSelf: base: activate:
+                  pkgs.buildEnv {
+                    name = ("activatable-" + base.name);
+                    paths =
+                      let
+                        hasDryActivate = builtins.hasAttr "dryActivate" customSelf;
+                      in [
+                        base
+                        (pkgs.writeTextFile {
+                          name = base.name + "-activate-path";
+                          text = ''
+                            #!${pkgs.runtimeShell}
+                            set -euo pipefail
+
+                            if [[ $DRY_ACTIVATE == "1" ]]
+                            then
+                                if ${pkgs.lib.boolToString hasDryActivate}
+                                then
+                                    ${if hasDryActivate then customSelf.dryActivate else ":"}
```

Comment:
Like @balsoft said, it might be nicer to have it print the contents of the activation script as a default instead of just noop.